diff -ruN '--exclude=.cargo-ok' '--exclude=Cargo.lock' '--exclude=.cargo_vcs_info.json' '--exclude=Cargo.toml.orig' a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml	1970-01-01 01:00:01.000000000 +0100
+++ b/Cargo.toml	2026-01-10 19:52:55.999068966 +0100
@@ -51,6 +51,9 @@
 version = "1.19"
 features = ["full"]
 
+[dependencies.portable-atomic]
+version = "1"
+
 [dev-dependencies.async-global-executor]
 version = "2.3.1"
 
diff -ruN '--exclude=.cargo-ok' '--exclude=Cargo.lock' '--exclude=.cargo_vcs_info.json' '--exclude=Cargo.toml.orig' a/src/vnet/chunk.rs b/src/vnet/chunk.rs
--- a/src/vnet/chunk.rs	2006-07-24 03:21:28.000000000 +0200
+++ b/src/vnet/chunk.rs	2026-01-10 19:52:56.003468966 +0100
@@ -8,10 +8,11 @@
 use std::net::{IpAddr, SocketAddr};
 use std::ops::{BitAnd, BitOr};
 use std::str::FromStr;
-use std::sync::atomic::{AtomicU64, Ordering};
 use std::time::SystemTime;
+use portable_atomic::AtomicU64;
 
 lazy_static! {
+    // ARMv5TEJ doesn't support AtomicU64 natively, use portable-atomic instead
     static ref TAG_CTR: AtomicU64 = AtomicU64::new(0);
 }
 
@@ -34,7 +35,7 @@
 // Generate a base36-encoded unique tag
 // See: https://play.golang.org/p/0ZaAID1q-HN
 fn assign_chunk_tag() -> String {
-    let n = TAG_CTR.fetch_add(1, Ordering::SeqCst);
+    let n = TAG_CTR.fetch_add(1, portable_atomic::Ordering::SeqCst);
     base36(n)
 }
 
diff -ruN '--exclude=.cargo-ok' '--exclude=Cargo.lock' '--exclude=.cargo_vcs_info.json' '--exclude=Cargo.toml.orig' a/src/vnet/net.rs b/src/vnet/net.rs
--- a/src/vnet/net.rs	2006-07-24 03:21:28.000000000 +0200
+++ b/src/vnet/net.rs	2026-01-10 19:52:56.003468966 +0100
@@ -14,27 +14,28 @@
 use std::collections::HashMap;
 use std::net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr};
 use std::str::FromStr;
-use std::sync::atomic::{AtomicU64, Ordering};
 use std::sync::Arc;
 use tokio::net::UdpSocket;
 use tokio::sync::Mutex;
+use portable_atomic::AtomicU64;
 
 pub(crate) const LO0_STR: &str = "lo0";
 pub(crate) const UDP_STR: &str = "udp";
 
 lazy_static! {
+    // ARMv5TEJ doesn't support AtomicU64 natively, use portable-atomic instead
     pub static ref MAC_ADDR_COUNTER: AtomicU64 = AtomicU64::new(0xBEEFED910200);
 }
 
 pub(crate) type HardwareAddr = Vec<u8>;
 
 pub(crate) fn new_mac_address() -> HardwareAddr {
-    let b = MAC_ADDR_COUNTER
-        .fetch_add(1, Ordering::SeqCst)
-        .to_be_bytes();
+    let value = MAC_ADDR_COUNTER.fetch_add(1, portable_atomic::Ordering::SeqCst);
+    let b = value.to_be_bytes();
     b[2..].to_vec()
 }
 
+
 #[derive(Default)]
 pub(crate) struct VNetInternal {
     pub(crate) interfaces: Vec<Interface>,         // read-only
diff -ruN '--exclude=.cargo-ok' '--exclude=Cargo.lock' '--exclude=.cargo_vcs_info.json' '--exclude=Cargo.toml.orig' a/src/vnet/router.rs b/src/vnet/router.rs
--- a/src/vnet/router.rs	2006-07-24 03:21:28.000000000 +0200
+++ b/src/vnet/router.rs	2026-01-10 19:52:56.003468966 +0100
@@ -17,24 +17,26 @@
 use std::ops::{Add, Sub};
 use std::pin::Pin;
 use std::str::FromStr;
-use std::sync::atomic::{AtomicU64, Ordering};
 use std::sync::Arc;
 use std::time::SystemTime;
 use tokio::sync::{mpsc, Mutex};
 use tokio::time::Duration;
+use portable_atomic::AtomicU64;
 
 const DEFAULT_ROUTER_QUEUE_SIZE: usize = 0; // unlimited
 
 lazy_static! {
+    // ARMv5TEJ doesn't support AtomicU64 natively, use portable-atomic instead
     pub static ref ROUTER_ID_CTR: AtomicU64 = AtomicU64::new(0);
 }
 
 // Generate a unique router name
 fn assign_router_name() -> String {
-    let n = ROUTER_ID_CTR.fetch_add(1, Ordering::SeqCst);
+    let n = ROUTER_ID_CTR.fetch_add(1, portable_atomic::Ordering::SeqCst);
     format!("router{}", n)
 }
 
+
 // RouterConfig ...
 #[derive(Default)]
 pub struct RouterConfig {
