diff -ruN '--exclude=.cargo-ok' '--exclude=Cargo.lock' '--exclude=.cargo_vcs_info.json' '--exclude=Cargo.toml.orig' a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml	1970-01-01 01:00:01.000000000 +0100
+++ b/Cargo.toml	2026-01-10 19:52:55.985868966 +0100
@@ -58,6 +58,9 @@
 version = "1.19"
 features = ["full"]
 
+[dependencies.portable-atomic]
+version = "1"
+
 [dependencies.util]
 version = "0.7.0"
 features = ["conn"]
diff -ruN '--exclude=.cargo-ok' '--exclude=Cargo.lock' '--exclude=.cargo_vcs_info.json' '--exclude=Cargo.toml.orig' a/src/association/association_stats.rs b/src/association/association_stats.rs
--- a/src/association/association_stats.rs	2006-07-24 03:21:28.000000000 +0200
+++ b/src/association/association_stats.rs	2026-01-10 19:52:55.990268966 +0100
@@ -1,6 +1,7 @@
-use std::sync::atomic::{AtomicU64, Ordering};
+// ARMv5TEJ doesn't support AtomicU64 natively, use portable-atomic instead
+use portable_atomic::AtomicU64;
 
-#[derive(Default, Debug)]
+#[derive(Debug)]
 pub(crate) struct AssociationStats {
     n_datas: AtomicU64,
     n_sacks: AtomicU64,
@@ -9,52 +10,64 @@
     n_fast_retrans: AtomicU64,
 }
 
+impl Default for AssociationStats {
+    fn default() -> Self {
+        AssociationStats {
+            n_datas: AtomicU64::new(0),
+            n_sacks: AtomicU64::new(0),
+            n_t3timeouts: AtomicU64::new(0),
+            n_ack_timeouts: AtomicU64::new(0),
+            n_fast_retrans: AtomicU64::new(0),
+        }
+    }
+}
+
 impl AssociationStats {
     pub(crate) fn inc_datas(&self) {
-        self.n_datas.fetch_add(1, Ordering::SeqCst);
+        self.n_datas.fetch_add(1, portable_atomic::Ordering::SeqCst);
     }
 
     pub(crate) fn get_num_datas(&self) -> u64 {
-        self.n_datas.load(Ordering::SeqCst)
+        self.n_datas.load(portable_atomic::Ordering::SeqCst)
     }
 
     pub(crate) fn inc_sacks(&self) {
-        self.n_sacks.fetch_add(1, Ordering::SeqCst);
+        self.n_sacks.fetch_add(1, portable_atomic::Ordering::SeqCst);
     }
 
     pub(crate) fn get_num_sacks(&self) -> u64 {
-        self.n_sacks.load(Ordering::SeqCst)
+        self.n_sacks.load(portable_atomic::Ordering::SeqCst)
     }
 
     pub(crate) fn inc_t3timeouts(&self) {
-        self.n_t3timeouts.fetch_add(1, Ordering::SeqCst);
+        self.n_t3timeouts.fetch_add(1, portable_atomic::Ordering::SeqCst);
     }
 
     pub(crate) fn get_num_t3timeouts(&self) -> u64 {
-        self.n_t3timeouts.load(Ordering::SeqCst)
+        self.n_t3timeouts.load(portable_atomic::Ordering::SeqCst)
     }
 
     pub(crate) fn inc_ack_timeouts(&self) {
-        self.n_ack_timeouts.fetch_add(1, Ordering::SeqCst);
+        self.n_ack_timeouts.fetch_add(1, portable_atomic::Ordering::SeqCst);
     }
 
     pub(crate) fn get_num_ack_timeouts(&self) -> u64 {
-        self.n_ack_timeouts.load(Ordering::SeqCst)
+        self.n_ack_timeouts.load(portable_atomic::Ordering::SeqCst)
     }
 
     pub(crate) fn inc_fast_retrans(&self) {
-        self.n_fast_retrans.fetch_add(1, Ordering::SeqCst);
+        self.n_fast_retrans.fetch_add(1, portable_atomic::Ordering::SeqCst);
     }
 
     pub(crate) fn get_num_fast_retrans(&self) -> u64 {
-        self.n_fast_retrans.load(Ordering::SeqCst)
+        self.n_fast_retrans.load(portable_atomic::Ordering::SeqCst)
     }
 
     pub(crate) fn reset(&self) {
-        self.n_datas.store(0, Ordering::SeqCst);
-        self.n_sacks.store(0, Ordering::SeqCst);
-        self.n_t3timeouts.store(0, Ordering::SeqCst);
-        self.n_ack_timeouts.store(0, Ordering::SeqCst);
-        self.n_fast_retrans.store(0, Ordering::SeqCst);
+        self.n_datas.store(0, portable_atomic::Ordering::SeqCst);
+        self.n_sacks.store(0, portable_atomic::Ordering::SeqCst);
+        self.n_t3timeouts.store(0, portable_atomic::Ordering::SeqCst);
+        self.n_ack_timeouts.store(0, portable_atomic::Ordering::SeqCst);
+        self.n_fast_retrans.store(0, portable_atomic::Ordering::SeqCst);
     }
 }
